packages/astro/src/content/mutable-data-store.ts

```diff
@@ -404,13 +405,26 @@ export default new Map([\n${lines.join(',\n')}]);
 		if (!this.#dirty) {
 			return;
 		}
-		if (!this.#file) {
+		if (!this.#manifestFile || !this.#dir) {
 			throw new AstroError(AstroErrorData.UnknownFilesystemError);
 		}
 		try {
 			// Mark as clean before writing to disk so that it catches any changes that happen during the write
 			this.#dirty = false;
-			await this.#writeFileAtomic(this.#file, this.toString());
+
+			const manifest: Record<string, string[]> = {};
+
+			for (const [collectionName, entries] of this._collections) {
+				manifest[collectionName] = [];
+				const chunkedEntries = splitStringBySizeLimit(devalue.stringify(entries), CHUNK_SIZE_LIMIT);
```

Comment:
Rather than splitting the generated string I think it would be better to split on entries, and stringify them individually. That avoids issues of needing to generate very large strings